inc/features/blocks/personalization/edit.js

```diff
@@ -0,0 +1,351 @@
+import React, { Fragment, useState } from 'react';
+
+const { AudiencePicker } = Altis.Analytics.components;
+
+const {
+	BlockControls,
+	InnerBlocks,
+	InspectorControls,
+} = wp.blockEditor;
+const { createBlock, cloneBlock } = wp.blocks;
+const {
+	Button,
+	IconButton,
+	PanelBody,
+	Toolbar,
+} = wp.components;
+const { compose } = wp.compose;
+const { withSelect, withDispatch, useSelect } = wp.data;
+const { __ } = wp.i18n;
+
+/**
+ * Only variants can be direct descendents so that we can generate
+ * usable markup.
+ */
+const ALLOWED_BLOCKS = [ 'altis/personalization-variant' ];
+
+/**
+ * Start with a default template of one variant.
+ */
+const TEMPLATE = [
+	[ 'altis/personalization-variant' ],
+];
+
+// Component for fetching and displaying the variant title string.
+const VariantTitle = ( { variant } ) => {
+	const audience = useSelect( select => {
+		return select( 'audience' ).getPost( variant.attributes.audience );
+	}, [ variant.attributes.audience ] );
+	const isLoading = useSelect( select => {
+		return select( 'audience' ).getIsLoading();
+	}, [] );
+
+	if ( variant.attributes.fallback ) {
+		return __( 'Fallback', 'altis-experiments' );
+	}
+
+	if ( ! variant.attributes.audience ) {
+		return __( 'Select audience', 'altis-experiments' );
+	}
+
+	if ( audience && audience.title && audience.title.rendered ) {
+		return audience.title.rendered;
+	}
+
+	if ( isLoading ) {
+		return __( 'Loading...', 'altis-experiments' );
+	}
+
+	return '';
+}
+
+// Audience picker input.
+const Edit = ( {
+	addVariant,
+	attributes,
+	className,
+	clientId,
+	copyVariant,
```

Comment:
These callbacks should be called `on...` per style guide

Reply:
I don't agree here, these aren't callbacks they're actions called from within callbacks eg. `onClick={ () => copyVariant() }`. It'd be like having `onClick={ () => this.onSetState( {} ) }` in React as opposed to `onClick={ () => this.setState() }`.

Reply:
I think that's a false equivalency; the equivalent would be `onClick={ () => this.props.onSetState( {} ) }` which *would* have the `on`. These are parameters (props) to the component, not intrinsic properties of it, so should follow the prop rules.